### ISSUE TITLE
Harmony shimmer failing in some cases

### ISSUE THREAD

**Issue by tobeyStraitjacket** [SubnauticaModding/QModManager#176]
**Describe the bug**
Some users are reporting an error stopping some mods from being loaded. Issue seems to be coming from our Harmony shim, log output as follows. 

```
[Debug] [Info   :QMMHarmonyShimmer] Shimming QMods...
[Debug] [Error  :QMMHarmonyShimmer] Failed to shim 0Harmony.dll
[Debug] [Error  :QMMHarmonyShimmer] System.BadImageFormatException: Format of the executable (.exe) or library (.dll) is invalid.
  at Mono.Cecil.PE.ImageReader.ReadImage () [0x00013] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.PE.ImageReader.ReadImage (Mono.Disposable`1[T] stream, System.String file_name) [0x00007] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (Mono.Disposable`1[T] stream, System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00006] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (System.IO.Stream stream, Mono.Cecil.ReaderParameters parameters) [0x00018] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (System.IO.Stream stream) [0x00007] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.AssemblyDefinition.ReadAssembly (System.IO.Stream stream) [0x00000] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at QModManager.QMMHarmonyShimmer.QMMHarmonyShimmer.ShimQMods () [0x00179] in <2c45d767c91a491984d51c42609253d7>:0
[Debug] [Error  :QMMHarmonyShimmer] Failed to shim YamlDotNet.dll
[Debug] [Error  :QMMHarmonyShimmer] System.ArgumentOutOfRangeException: Non-negative number required.
Parameter name: count
  at System.IO.BinaryReader.ReadBytes (System.Int32 count) [0x00013] in <567df3e0919241ba98db88bec4c6696f>:0 
  at Mono.Cecil.PE.ImageReader.ReadDebugHeader () [0x000f7] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.PE.ImageReader.ReadImage () [0x000a8] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.PE.ImageReader.ReadImage (Mono.Disposable`1[T] stream, System.String file_name) [0x00007] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (Mono.Disposable`1[T] stream, System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00006] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (System.IO.Stream stream, Mono.Cecil.ReaderParameters parameters) [0x00018] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.ModuleDefinition.ReadModule (System.IO.Stream stream) [0x00007] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at Mono.Cecil.AssemblyDefinition.ReadAssembly (System.IO.Stream stream) [0x00000] in <6034b380a22b41a596c9dc29d282c0a9>:0 
  at QModManager.QMMHarmonyShimmer.QMMHarmonyShimmer.ShimQMods () [0x00179] in <2c45d767c91a491984d51c42609253d7>:0
```

This seems to be happening when the assembly being looked at is not in the correct format (probably unmanaged), and is often also happening when some mod authors are distributing their own version of 0Harmony.dll with their mod (oof - we should probably make it clear in the docs that this is a VERY BAD IDEA).

Posting this here as a reminder for myself to look into this ASAP.

**Comment by PrimeSonic** [comment]
Noting that the only remaining mods doing this are very old mods that are no longer actively maintained.

**Comment by tobeyStraitjacket** [comment]
Indeed, this will be closed by the added logging in #179
